internal/config/action.go

```diff
@@ -0,0 +1,52 @@
+package config
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+type Action int
+
+const (
+	InvalidAction Action = iota
+	UpgradeAction
+	DryRunAction
+	CheckUpToDateAction
+	CheckDeactivatedPlansAction
+	MinVersionCheckAction
+)
+
+func determineAction(checkDeactivatedPlans, checkUpToDate, dryRun bool, minVersionRequired string) (Action, error) {
+	spec := map[string]bool{
```

Comment:
praise: The use of a map `spec` to track which flags are set is a clever approach. It simplifies the process of determining which flags are active and helps in generating error messages.